[issue-115] Corrected the notation of isotopes in the node descriptio…

imbeauf opened the pull request:
…ns in DISTRIBUTIONS, DISTRIBUTIONS_SOURCES and SPECTROMETER_VISIBLE IDS, and also in the DISTRIBUTION_SOURCE_IDENTIFIER names
closes #115 

<!-- readthedocs-preview imas-data-dictionary start -->
----
📚 Documentation preview 📚: https://imas-data-dictionary--133.org.readthedocs.build/en/133/

<!-- readthedocs-preview imas-data-dictionary end -->

imbeauf commented:
@DavidPCoster , could you take a look at this PR, please ?

maarten-ic commented on schemas/utilities/dd_support.xsd:
```diff
@@ -5006,7 +5006,7 @@
 			<xs:element name="type" type="identifier">
 				<xs:annotation>
-					<xs:documentation>Process type. index=1 for NBI; index=2 for nuclear reaction (reaction unspecified); index=3 for nuclear reaction: T(d,n)4He [D+T->He4+n]; index=4 for nuclear reaction: He3(d,p)4He [He3+D->He4+p]; index=5 for nuclear reaction: D(d,p)T [D+D->T+p]; index=6 for nuclear reaction: D(d,n)3He [D+D->He3+n]; index=7 for runaway processes</xs:documentation>
+					<xs:documentation>Process type. index=1 for NBI; index=2 for nuclear reaction (reaction unspecified); index=3 for nuclear reaction: T(d,n)4He [D+T->4He+n]; index=4 for nuclear reaction: 3He(d,p)4He [3He+D->4He+p]; index=5 for nuclear reaction: D(d,p)T [D+D->T+p]; index=6 for nuclear reaction: D(d,n)3He [D+D->3He+n]; index=7 for runaway processes</xs:documentation>
```
Please note that this description doesn't match with the index values provided by the identifier XML.

MireilleSchneider replied:
Note that if this is just about changing the values of the identifiers, it should actually be transparent, since we use the al_distribution_source_identifier package.

imbeauf replied:
Ok so I will change the description of the DD node and keep the identifier indices as they are, thanks !

janthomasjohnson replied:
As @MireilleSchneider said, ASCOT, BBNBI and other HCD codes from the EU use the derived types, `distribution_source_identifier%*`., in the same way as done in NEMO. So, I don't think this will cause an issue, unless we're receiving data from an older DD on input.

olivhoenen commented:
> In link with discussion #138 , I have introduced an alias1 metadata in the modified identifiers.

Thanks. Few questions on this:
- Why the attribute is named `alias1` and not `alias`? do you foresee further renaming (hopefully we don't get there) to introduce an `alias2`? would it not be better to get a list of `alias="oldname1,oldname2"`?
- It may be good to capture this new construct somewhere in the documentation (e.g. in https://imas-data-dictionary.readthedocs.io/en/latest/identifiers.html)
- The alias info is not exposed until one look at the DD sources, it may be good to have it in tables (e.g. https://imas-data-dictionary--133.org.readthedocs.build/en/133/generated/identifier/materials_identifier.html)

imbeauf commented:
> > In link with discussion #138 , I have introduced an alias1 metadata in the modified identifiers.
> 
> Thanks. Few questions on this:
> 
> * Why the attribute is named `alias1` and not `alias`? do you foresee further renaming (hopefully we don't get there) to introduce an `alias2`? would it not be better to get a list of `alias="oldname1,oldname2"`?
> * It may be good to capture this new construct somewhere in the documentation (e.g. in https://imas-data-dictionary.readthedocs.io/en/latest/identifiers.html)
> * The alias info is not exposed until one look at the DD sources, it may be good to have it in tables (e.g. https://imas-data-dictionary--133.org.readthedocs.build/en/133/generated/identifier/materials_identifier.html)

Point 1: ok, I have changed to alias + a comma-separated list
Point 2: yes, I have updated the DD developer guide with this new feature
Point 3: we'll need Maarten to implement the display feature
